tests/python/rest_api/test_load_register.py

```diff
@@ -0,0 +1,150 @@
+from locust.stats import stats_history, stats_printer
+
+from shared.utils.config import ASSETS_DIR, BASE_URL, USER_PASS
+
+
+class AdminUserNotFoundInDbException(Exception):
+    def __init__(self):
+        print("Can't find any admin user in the test DB")
+
+
+class Container:
+    """
+    We have to recreate the fixtures' data generation,
+    since locust doesn't support using pytest tests as tasks yet
+    """
+
+    def __init__(self, data, key="id"):
+        self.raw_data = data
+        self.map_data = {obj[key]: obj for obj in data}
+
+    @property
+    def raw(self):
+        return self.raw_data
+
+    @property
+    def map(self):
+        return self.map_data
+
+    def __iter__(self):
+        return iter(self.raw_data)
+
+    def __len__(self):
+        return len(self.raw_data)
+
+    def __getitem__(self, key):
+        if isinstance(key, slice):
+            return self.raw_data[key]
+        return self.map_data[key]
+
+
+class GetUser:
+    """
+    We have to recreate the fixtures' data generation,
+    since locust doesn't support using pytest tests as tasks yet
+    """
+
+    @staticmethod
+    def get_admin_user():
+        # getting the user from local db
+        with open(os.path.join(ASSETS_DIR, "users.json")) as f:
+            container = Container(json.load(f)["results"])
+
+            for user in container:
+                if user["is_superuser"] and user["is_active"]:  # getting the admin user
+                    return user["username"]
+
+            raise AdminUserNotFoundInDbException()
+
+
+class PerformTask(locust.SequentialTaskSet):
+    @locust.task
+    def perform_random_user_register_task(self):
+        # Initializing Faker for data generation
+        fake = Faker()
+        username = None
+        first_name = fake.first_name()
+        last_name = fake.last_name()
+        username_type = bool(random.getrandbits(1))
+
+        if username_type == 0:
+            username = first_name + last_name
+        elif username_type == 1:
+            username = last_name + first_name
+
+        email = fake.ascii_email()
+
+        # Setting a random passwd
+        password_length = random.randrange(13, 24)
+        passwd = secrets.token_urlsafe(password_length)
+
+        # Setting the config for client auth
+        # Needs further dehardcoding
+        config = Configuration(
+            host=LoadUser.host, username=GetUser.get_admin_user(), password=USER_PASS
+        )
+
+        # Launching the client with randomized data
+        with ApiClient(configuration=config) as api_client:
+            try:
+                # forming the request body
+                register_request = RegisterSerializerExRequest(
+                    username=username,
+                    password1=passwd,
+                    password2=passwd,
+                    email=email,
+                    first_name=first_name,
+                    last_name=last_name,
+                )
```

Comment:
When I run the test locally, I see some requests fail with 400, but the test doesn't fail. I think we should implement this test in such a way that we don't have this error.

Reply:
Do you want to catch the request exceptions and make the test fail upon parsing a non-200 response, or to suppress errors when sending the request altogether?

Reply:
Ended up doing both

Reply:
In my opinion the should fail if server return non-200 responses.
We are trying to check the average user registration time, but how can we trust this number if some (or all) of our registrations fail.

Reply:
Sure, thought so too. Corrected as needed.